src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue25124.cs

```diff
@@ -0,0 +1,22 @@
+using NUnit.Framework;
+using UITest.Appium;
+using UITest.Core;
+
+namespace Microsoft.Maui.TestCases.Tests.Issues;
+
+public class Issue25124 : _IssuesUITest
+{
+	public Issue25124(TestDevice testDevice) : base(testDevice)
+	{
+	}
+
+	public override string Issue => "OnPlatform does not work in the Header and Footer of CollectionView";
+
+	[Test]
+	[Category(UITestCategories.CollectionView)]
+	public void ShouldDisplayHeaderBasedOnOnPlatform()
+	{
+		App.WaitForElement("CollectionView");
+		VerifyScreenshot();
```

Comment:
Pending snapshots:
<img width="361" alt="image" src="https://github.com/user-attachments/assets/b2b237c7-c0c8-4e42-a917-2a0ed62ec809" />

For example, Mac:
![image](https://github.com/user-attachments/assets/58fe661e-f94f-4961-ba16-1395d34b4253)

Could you commit it?

Reply:
Test failing on Android:
![image](https://github.com/user-attachments/assets/db9526a7-f879-4685-b821-b79a58e181ec)